changelogs/fragments/2454-detect_zfs_changed.yml

```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - zfs - Certain ZFS properties, especially sizes, would lead to a task being falsely marked as "changed" even when no actual change was made. Resolves https://github.com/ansible-collections/community.general/issues/975 (https://github.com/ansible-collections/community.general/pull/2454)
```

Comment:
```suggestion
  - zfs - certain ZFS properties, especially sizes, would lead to a task being falsely marked as "changed" even when no actual change was made (https://github.com/ansible-collections/community.general/issues/975, https://github.com/ansible-collections/community.general/pull/2454).
```